libdnf/system/state.hpp

```diff
@@ -210,6 +212,12 @@ class State {
     std::map<std::string, GroupState> group_states;
     std::map<std::string, ModuleState> module_states;
     SystemState system_state;
+
+    // Caches to speed-up searching the maps by values
+    //
+    // map caches groups for packages. Package name is the key, set of group ids is the value
+    const std::map<std::string, std::set<std::string>> & get_package_groups_cache();
```

Comment:
Just a nitpick, I'd prefer the method to be in a block with the other private methods and the member variables in a separate block below them. Unless you prefer this way for a reason...

Reply:
I've moved them. Originally there were two caches so I wanted to keep them together.